bin/fuel-core/src/cli/run.rs

```diff
@@ -210,10 +211,14 @@ pub struct Command {
     pub min_da_gas_price: u64,
 
     /// P component of DA gas price calculation
-    #[arg(long = "da-p-component", default_value = "0", env)]
+    /// **NOTE**: This is the **inverse** gain of a typical P controller.
+    /// Increasing this value will reduce gas price fluctuations.
+    #[arg(long = "da-p-component", default_value = "1", env)]
```

Comment:
Note that we're defaulting to maxing out the gain. This would result in spiky algorithm behavior for unconfigured nodes, which may be acceptable. I would ideally like this to be the smallest possible gain, but I fear a `i64::MAX` would likely result in no algo changes after discretization.

Note that this parameter only matters for the authority node, since any other node will use previous blocks to estimate gas prices as per https://github.com/FuelLabs/fuel-core/pull/2501

So the use case as I understand it is someone spinning up their own chain with default configuration, potentially tweaking other parameters.

Reply:
may I suggest that this PR should be tested with the SDK's which have some e2e tests which run fuel-core under the hood?
I don't think they change any of these values so it will be nice to see the new defaults in action and if it affects their tests

Reply:
Yeah I would love to get the link if you have it. I'll look into the tests tomorrow.

Reply:
> Note that we're defaulting to maxing out the gain. This would result in spiky algorithm behavior for unconfigured nodes, which may be acceptable. I would ideally like this to be the smallest possible gain, but I fear a i64::MAX would likely result in no algo changes after discretization.

I really appreciate this sort of feedback. I've thought about this a bunch, but I'd prefer to get different perspectives, rather than just guessing what would be the "right" behavior.

Reply:
Yeah, I don't see any obvious "smart" default values here since this would have to be tuned against other system parameters (max gas, block time etc.) in combination with the expected system load. So defaulting to the max value here would at least make sure we jump in the right direction - and rely on the change percent clamping to not go too far.

Reply:
Writing a test now, I think the `FuelCoreDriver` from the integration test helpers will be enough for this.

Reply:
After a lot of debugging I now have a working test in https://github.com/FuelLabs/fuel-core/pull/2507/commits/4d0108210995c15fd19719cf761ce9189688ec99. Because the default gas limit is pretty high, this test is a bit slow, and I am hesitant to whether the test adds any value. Otherwise I'm happy to remove it.